packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -512,69 +517,21 @@ class AnimationController extends Animation<double>
-          'The "duration" or "reverseDuration" property should be set, either in the constructor or later, before '
-          'calling the reverse() function.',
-        );
-      }
-      return true;
-    }());
-    assert(
-      _ticker != null,
-      'AnimationController.reverse() called after AnimationController.dispose()\n'
-      'AnimationController methods should not be used after calling dispose.',
-    );
-    _direction = _AnimationDirection.reverse;
-    if (from != null) {
-      value = from;
-    }
-    return _animateToInternal(lowerBound);
-  }
+  TickerFuture reverse({ double? from }) => _toggle('reverse', false, from);
 
-  /// Toggles the direction of this animation, based on whether it [isForwardOrCompleted].
+  /// Toggles the direction of this animation.
   ///
-  /// Specifically, this function acts the same way as [reverse] if the [status] is
-  /// either [AnimationStatus.forward] or [AnimationStatus.completed], and acts as
-  /// [forward] for [AnimationStatus.reverse] or [AnimationStatus.dismissed].
-  ///
-  /// If [from] is non-null, it will be set as the current [value] before running
-  /// the animation.
+  /// If [toggleForward] is non-null, it determines the animation direction.
+  /// Otherwise, the behavior is determined by [isForwardOrCompleted], animating
+  /// to the [lowerBound] if the [status] is either [AnimationStatus.forward] or
+  /// [AnimationStatus.completed], or the [upperBound] for
+  /// [AnimationStatus.reverse] or [AnimationStatus.dismissed].
   ///
   /// The most recently returned [TickerFuture], if any, is marked as having been
   /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
   /// derivative future completes with a [TickerCanceled] error.
-  TickerFuture toggle({ double? from }) {
-    assert(() {
-      Duration? duration = this.duration;
-      if (isForwardOrCompleted) {
-        duration ??= reverseDuration;
-      }
-      if (duration == null) {
-        throw FlutterError(
-          'AnimationController.toggle() called with no default duration.\n'
-          'The "duration" property should be set, either in the constructor or later, before '
-          'calling the toggle() function.',
-        );
-      }
-      return true;
-    }());
-    assert(
-      _ticker != null,
-      'AnimationController.toggle() called after AnimationController.dispose()\n'
-      'AnimationController methods should not be used after calling dispose.',
-    );
-    _direction = isForwardOrCompleted ? _AnimationDirection.reverse : _AnimationDirection.forward;
-    if (from != null) {
-      value = from;
-    }
-    return _animateToInternal(switch (_direction) {
-      _AnimationDirection.forward => upperBound,
-      _AnimationDirection.reverse => lowerBound,
-    });
+  TickerFuture toggle([ bool? toggleForward ]) {
+    return _toggle('toggle', toggleForward ?? !isForwardOrCompleted, null);
   }
```

Comment:
Here's the new method implementation!

I decided to add a private `_toggle()` method so we didn't need to have the same asserts in 3 different places :)